From e73d91ecca7c5d5ded22f94c9cc786c9b0cbee99 Mon Sep 17 00:00:00 2001
From: Iceows <mounierr07@gmail.com>
Date: Mon, 17 Nov 2025 19:44:31 +0100
Subject: [PATCH 01/19] huawei: remove bad sec_storage mount and fix some
 issues with ......

huawei: remove bad sec_storage mount and fix some issues with media, rpc motions sensor and ar
---
 rw-system.sh | 36 +++++++++++++++++-------------------
 1 file changed, 17 insertions(+), 19 deletions(-)

diff --git a/rw-system.sh b/rw-system.sh
index e752712..107c654 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -650,6 +650,20 @@ if getprop ro.vendor.build.fingerprint | grep -iq -E -e 'huawei|honor' || getpro
         mount -o bind /product/etc/libnfc-nxp.conf /system/etc/libnfc-nxp.conf || true
 
     mount -o bind /system/phh/libnfc-nci-huawei.conf /system/etc/libnfc-nci.conf
+
+    # Set proper /sdcard permissions to avoid dead storage
+    chown media_rw:media_rw /data/media/0
+    chmod 0770 /data/media/0
+
+    # Disable this watchlss extension to fix logspams and dead gps
+    mount -o bind /mnt/phh/empty /vendor/bin/gnss_watchlssd_thirdparty
+
+    # For audio_custom
+    chown system:system /sys/class/sensors/rpc_sensor/rpc_motion_req
+
+    # For activity reco
+    chmod 0644  /dev/ar
+    chown system:system /dev/ar
 fi
 
 if getprop ro.vendor.build.fingerprint | grep -qE -e ".*(crown|star)[q2]*lte.*" -e ".*(SC-0[23]K|SCV3[89]).*" && [ "$vndk" -lt 28 ]; then
@@ -805,6 +819,7 @@ if getprop ro.boot.boot_devices |grep -v , |grep -qE .;then
     ln -s /dev/block/platform/$(getprop ro.boot.boot_devices) /dev/block/bootdevice
 fi
 
+# Huawei device
 if [ -c /dev/dsm ];then
     # /dev/dsm is a magic device on Kirin chipsets that teecd needs to access.
     # Make sure that permissions are right.
@@ -812,25 +827,8 @@ if [ -c /dev/dsm ];then
     chmod 0660 /dev/dsm
 
     # The presence of /dev/dsm indicates that we have a teecd,
-    # which needs /sec_storage and /data/sec_storage_data
-
-    mkdir -p /data/sec_storage_data
-    chown system:system /data/sec_storage_data
-    chcon -R u:object_r:teecd_data_file:s0 /data/sec_storage_data
-
-    if mount | grep -q " on /sec_storage " ; then
-        # /sec_storage is already mounted by the vendor, don't try to create and mount it
-        # ourselves. However, some devices have /sec_storage owned by root, which means that
-        # the fingerprint daemon (running as system) cannot access it.
-        chown -R system:system /sec_storage
-        chmod -R 0660 /sec_storage
-        chcon -R u:object_r:teecd_data_file:s0 /sec_storage
-    else
-        # No /sec_storage provided by vendor, mount /data/sec_storage_data to it
-        mount /data/sec_storage_data /sec_storage
-        chown system:system /sec_storage
-        chcon u:object_r:teecd_data_file:s0 /sec_storage
-    fi
+    # which needs /sec_storage and /data/sec_storage_data provide by init.huawei.os.a15.rc
+
 fi
 
 #Weird /odm/phone.prop Huawei stuff
-- 
2.25.1

